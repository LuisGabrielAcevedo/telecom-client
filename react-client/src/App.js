import React from "react";
import "./App.scss";
import CurrentWeather from "./components/CurrentWeather";
import CitiesWeather from "./components/CitiesWeather";

function App() {
  return (
    <div>
      <CurrentWeather />
      <CitiesWeather />
    </div>
  );
}

export default App;
