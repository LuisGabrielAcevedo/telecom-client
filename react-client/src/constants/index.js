export const BASE_URL = "https://api.openweathermap.org/data/2.5";
export const APP_KEY = "b8765b5f6bc11600d4df5c2b21caa611";

export const countries = [
  { label: "Madrid", value: "Madrid" },
  { label: "Paris", value: "Paris" },
  { label: "Praga", value: "Praga" },
  { label: "Sao Paulo", value: "Sao Paulo" },
  { label: "Lima", value: "Lima" },
];

export const selectLabels = {
  selectSomeItems: "Seleccione...",
  allItemsAreSelected: "Todos las cuidades estan seleccionados",
  selectAll: "Seleccione todas las ciudades",
  search: "Buscar",
};
