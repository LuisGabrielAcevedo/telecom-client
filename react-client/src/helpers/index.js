import {
  getWeatherByPositionApi,
  getForecastByPositionApi,
  getWeatherByCityApi,
  getForecastByCityApi,
} from "src/api";

export const getNextDaysWeatherByPosition = async (lat, lon) => {
  const resp = await getForecastByPositionApi(lat, lon);
  resp.list = resp.list.filter((i) => i.dt_txt.includes("00:00:00"));
  return resp;
};

export const getCurrentWeather = async (lat, lon) => {
  const resp = await Promise.all([
    getWeatherByPositionApi(lat, lon),
    getNextDaysWeatherByPosition(lat, lon),
  ]);

  const data = {
    title: resp[0].name,
    temp: resp[0].main.temp,
    wind: resp[0].wind.speed,
    humidity: resp[0].main.humidity,
    weather: resp[0].weather[0].main,
    nextDays: resp[1].list,
  };

  return data;
};

export const getNextDaysWeather = async (city) => {
  const resp = await getForecastByCityApi(city);
  resp.list = resp.list.filter((i) => i.dt_txt.includes("00:00:00"));
  return resp;
};

export const getWeatherByCity = async (selected) => {
  const values = selected.map((t) => t.value);
  const preWeatherPromise = [];
  const preNextDaysWeatherPromise = [];
  values.forEach((v) => {
    preWeatherPromise.push(getWeatherByCityApi(v));
    preNextDaysWeatherPromise.push(getNextDaysWeather(v));
  });
  const weatherResp = await Promise.all(preWeatherPromise);
  const nextDaysWeatherResp = await Promise.all(preNextDaysWeatherPromise);

  const data = weatherResp.map((resp, i) => {
    return {
      title: resp.name,
      temp: resp.main.temp,
      wind: resp.wind.speed,
      humidity: resp.main.humidity,
      weather: resp.weather[0].main,
      nextDays: nextDaysWeatherResp[i].list,
    };
  });
  return data;
};
