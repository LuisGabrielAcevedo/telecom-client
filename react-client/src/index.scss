@import "./assets/variables.scss";

html {
  background-color: $color-quinary;
  padding: $spacing-M;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}
