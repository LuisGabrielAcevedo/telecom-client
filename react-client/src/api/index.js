import { BASE_URL, APP_KEY } from "../constants";

export const getWeatherByPositionApi = async (lat, lon) => {
  try {
    const resp = await fetch(
      `${BASE_URL}/weather?lat=${lat}&lon=${lon}&appid=${APP_KEY}`
    );
    return resp.json();
  } catch (err) {
    return err;
  }
};

export const getForecastByPositionApi = async (lat, lon) => {
  try {
    const resp = await fetch(
      `${BASE_URL}/forecast?lat=${lat}&lon=${lon}&appid=${APP_KEY}`
    );
    return resp.json();
  } catch (err) {
    return err;
  }
};

export const getWeatherByCityApi = async (city) => {
  try {
    const resp = await fetch(`${BASE_URL}/weather?q=${city}&appid=${APP_KEY}`, {
      method: "GET",
    });
    return resp.json();
  } catch (err) {
    return err;
  }
};

export const getForecastByCityApi = async (city) => {
  try {
    const resp = await fetch(`${BASE_URL}/forecast?q=${city}&appid=${APP_KEY}`);
    return resp.json();
  } catch (err) {
    return err;
  }
};
