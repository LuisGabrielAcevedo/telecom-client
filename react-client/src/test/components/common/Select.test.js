import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import Select from "src/components/common/Select";
import { countries } from "src/constants";

describe("Select Component test", () => {
  const selected = { label: "Madrid", value: "Madrid" };
  const setSelected = jest.fn();
  const wrapper = shallow(
    <Select selected={selected} setSelected={setSelected} />
  );

  it("the component should be initialized", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("when the MultiSelect child component changes should call the setSelected prop", () => {
    const select = wrapper.find("MultiSelect");
    select.prop("onChange")();
    expect(setSelected).toHaveBeenCalled();
  });

  it("when the component is mounted the MultiSelect child component should has the countries", () => {
    const select = wrapper.find("MultiSelect");
    const options = select.prop("options");
    expect(options).toEqual(countries);
  });
});
