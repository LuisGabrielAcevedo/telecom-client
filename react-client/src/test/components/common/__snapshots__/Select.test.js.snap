// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component test the component should be initialized 1`] = `
<div
  style={
    Object {
      "width": "800px",
    }
  }
>
  <h1>
    Seleccione las ciudades para buscar el clima:
  </h1>
  <MultiSelect
    labelledBy="Ciudades"
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Madrid",
          "value": "Madrid",
        },
        Object {
          "label": "Paris",
          "value": "Paris",
        },
        Object {
          "label": "Praga",
          "value": "Praga",
        },
        Object {
          "label": "Sao Paulo",
          "value": "Sao Paulo",
        },
        Object {
          "label": "Lima",
          "value": "Lima",
        },
      ]
    }
    overrideStrings={
      Object {
        "allItemsAreSelected": "Todos las cuidades estan seleccionados",
        "search": "Buscar",
        "selectAll": "Seleccione todas las ciudades",
        "selectSomeItems": "Seleccione...",
      }
    }
    value={
      Object {
        "label": "Madrid",
        "value": "Madrid",
      }
    }
  />
</div>
`;
